### PR TITLE
python-language-server: 2021-05-20 -> 2021-09-08, use buildDotnetModule

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
Done both for easier maintage, and consistency

cc @thomasjm

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by thomasjm** [comment]
Oh nice, this looks great. Thanks! Did you test all the common LSP features like hover, completion, etc?

**Comment by IvarWithoutBones** [comment]
Yes, everything seems to work just like it should :)

File: pkgs/development/dotnet-modules/python-language-server/updater.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p nuget-to-nix dotnet-sdk_3
+set -eo pipefail
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+deps_file="$(realpath "./deps.nix")"
+
+cd ../../../..
```
Comment (thomasjm) [inline pkgs/development/dotnet-modules/python-language-server/updater.sh:8]: Maybe use `pushd` here for consistency?

File: pkgs/development/dotnet-modules/python-language-server/updater.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+chmod -R +w "$src"
+
+pushd "$src"
+
+mkdir ./nuget_tmp.packages
+dotnet restore src/LanguageServer/Impl/Microsoft.Python.LanguageServer.csproj --packages ./nuget_tmp.packages
+
+nuget-to-nix ./nuget_tmp.packages > "$deps_file"
+
+popd
```
Comment (thomasjm) [inline pkgs/development/dotnet-modules/python-language-server/updater.sh:21]: Unnecessary `popd`

**Review by thomasjm — APPROVED** [review]
I haven't had time to test it myself yet but I'll try and get to it in the next few days. Code-wise I had just a couple nits. I'm a big fan of the simplification here :)

**Comment by IvarWithoutBones** [comment]
I apologize about taking such a long time to adress your review, but it should be good now :)

File: pkgs/development/dotnet-modules/python-language-server/updater.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+nix-prefetch-git https://github.com/microsoft/python-language-server --quiet > repo_info
+new_version="$(jq -r ".date" < repo_info | cut -d"T" -f1)"
+new_hash="$(jq -r ".sha256" < repo_info)"
+new_rev="$(jq -r ".rev" < repo_info)"
+rm repo_info
+
+old_rev="$(sed -nE 's/\s*rev = "(.*)".*/\1/p' ./default.nix)"
+
+if [[ "$new_rev" == "$old_rev" ]]; then
```
Comment (SuperSandro2000) [inline pkgs/development/dotnet-modules/python-language-server/updater.sh:16]:
```suggestion
if [[ $new_rev == $old_rev ]]; then
```
shellcheck would have a problem with this: quoting both sides has strange side effects.

Comment (IvarWithoutBones) [reply pkgs/development/dotnet-modules/python-language-server/updater.sh]: Done :+1:

**Comment by IvarWithoutBones** [comment]
Is anything blocking this @thomasjm @SuperSandro2000?

**Comment by thomasjm** [comment]
It looks good to me! I say go ahead and merge.

**Pull request merged**